operations/configuration_management/Management_Node_Image_Customization.md

```diff
@@ -423,57 +429,80 @@ new customized image.
 
 1. (`ncn-mw#`) Update boot parameters for a management node.
 
-    Perform the following procedure **for each xname** identified in the previous step.
-
-    1. Get the existing `metal.server` setting for the component name (xname) of the node of interest.
+    * When working with a group of `XNAMES` (use the previous step to collect the group of `XNAMES`).
 
         ```bash
-        XNAME=<node-xname>
-        METAL_SERVER=$(cray bss bootparameters list --hosts "${XNAME}" --format json | jq '.[] |."params"' \
+        for XNAME in "${XNAMES[@]}"; do
+          echo "--- $XNAME ---"
+          METAL_SERVER=$(cray bss bootparameters list --hosts "${XNAME}" --format json | jq '.[] |."params"' \
             | awk -F 'metal.server=' '{print $2}' \
             | awk -F ' ' '{print $1}')
-        echo "${METAL_SERVER}"
+          echo "current metal.server: ${METAL_SERVER}"
+          S3_ARTIFACT_PATH="boot-images/${NEW_IMS_IMAGE_ID}"
+          NEW_METAL_SERVER="s3://${S3_ARTIFACT_PATH}/rootfs"
+          echo "    new metal.server: ${NEW_METAL_SERVER}"
+          PARAMS=$(cray bss bootparameters list --hosts "${XNAME}" --format json | jq '.[] |."params"' | \
+            sed "/metal.server/ s|${METAL_SERVER}|${NEW_METAL_SERVER}|" | \
+            sed "s/metal.no-wipe=1/metal.no-wipe=0/" | \
+            tr -d \")
+          cray bss bootparameters update --hosts "${XNAME}" \
+            --kernel "s3://${S3_ARTIFACT_PATH}/kernel" \
+            --initrd "s3://${S3_ARTIFACT_PATH}/initrd" \
+            --params "${PARAMS}"
+        done
         ```
 
-    1. Create updated boot parameters that point to the new artifacts.
-
-        1. Set the path to the artifacts in S3.
+    * When working on one node at a time.
 
-            **NOTE:** This uses the `NEW_IMS_IMAGE_ID` variable set in an earlier step.
+        1. Get the existing `metal.server` setting for the component name (xname) of the node of interest.
 
             ```bash
-            S3_ARTIFACT_PATH="boot-images/${NEW_IMS_IMAGE_ID}"
-            echo "${S3_ARTIFACT_PATH}"
+            XNAME=<node-xname>
```

Comment:
Is there a reason that the once the XNAME is set here, you did not use the single block of code that was used in a loop above to execute all these steps together?

Reply:
I had left the original steps alone, which were for running against one node at a time.
They could be enhanced, I think it would be better/simpler if they were.

Reply:
Makes sense. It works as is. We can make this change in a new PR that will go into all the branches.

Reply:
Opting for that, marking this as resolved.